SourceCode/AgOpenGPS.Core/Models/Guidance/DubinsPath.cs

```diff
@@ -0,0 +1,265 @@
+            GoalTurnType = goalTurnType;
+            _isComputed = false;
+        }
+
+        public TurnType StartTurnType { get; }
+        public TurnType MiddleTurnType { get; }
+        public TurnType GoalTurnType { get; }
+
+        public double TotalLength
+        {
+            get
+            {
+                Compute();
+                return _totalLength;
+            }
+        }
+
+        public double Length1
+        {
+            get
+            {
+                Compute();
+                return _length1;
+            }
+        }
+
+        public double Length2
+        {
+            get
+            {
+                Compute();
+                return _length2;
+            }
+        }
+
+        public double Length3
+        {
+            get
+            {
+                Compute();
+                return _length3;
+            }
+        }
+
+        private void Compute()
+        {
+            if (_isComputed) return;
+            _isComputed = true;
+            _startCircle = ComputeCircle(_constraints.StartConstraint, _constraints.RadiusConstraint, StartTurnType);
+            _goalCircle = ComputeCircle(_constraints.GoalConstraint, _constraints.RadiusConstraint, GoalTurnType);
+            ComputeTangents(_startCircle, StartTurnType, _goalCircle, out _startTangent, out _goalTangent);
+            _length1 = _startCircle.GetArcLength(_constraints.StartConstraint.Coord, _startTangent, StartTurnType);
+            _length2 = MiddleTurnType == TurnType.Straight ?
+                (_goalTangent - _startTangent).Length :
+                _middleCircle.GetArcLength(_startTangent, _goalTangent, MiddleTurnType);
+            _length3 = _goalCircle.GetArcLength(_goalTangent, _constraints.GoalConstraint.Coord, GoalTurnType);
+            _totalLength = _length1 + _length2 + _length3;
+        }
+
+        public static GeoCircle ComputeCircle(GeoCoordDir cdOnCircle, double radius, TurnType turnType)
+        {
+            Debug.Assert(turnType != TurnType.Straight);
+            GeoDir dir = turnType == TurnType.Right ? cdOnCircle.Direction.PerpendicularRight : cdOnCircle.Direction.PerpendicularLeft;
+            return new GeoCircle(cdOnCircle.Coord + radius * dir, radius);
+        }
+    }
+
+    // Base class of LslDubinsPath and RsrDubinsPath
+    public abstract class OuterDubinsPath : DubinsPath
+    {
+        public OuterDubinsPath(
+            DubinsPathConstraints constraints,
+            TurnType startTurnType, TurnType goalTurnType) : base(constraints, startTurnType, TurnType.Straight, goalTurnType)
+        {
+            Debug.Assert(startTurnType != TurnType.Straight);
+            Debug.Assert(goalTurnType != TurnType.Straight);
+        }
+
+        static public bool PathIsPossible(GeoCircle startCircle, GeoCircle goalCircle)
+        {
+            bool isPossible = startCircle.Center.Easting != goalCircle.Center.Easting && startCircle.Center.Northing != goalCircle.Center.Northing;
```

Comment:
Shouldn't this be "or":
```suggestion
            bool isPossible = startCircle.Center.Easting != goalCircle.Center.Easting || startCircle.Center.Northing != goalCircle.Center.Northing;
```
?

Reply:
A bug indeed.